src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -309,6 +291,151 @@ private string GetField(FieldExpression f)
         {
             return _query.Metadata.GetIndexFieldName(f, _query.QueryParameters).Value;
         }
+        
+        private IQueryMatch OrderBy(IQueryMatch match, List<(QueryExpression Expression, OrderByFieldType FieldType, bool Ascending)> orders, int take)
+        {
+            switch (orders.Count)
+            {
+                //Note: we want to use generics up to 3 comparers. This way we gonna avoid virtual calls in most cases.
+                case 1:
+                {
+                    if (orders[0].Expression is not FieldExpression fe)
+                        throw new InvalidQueryException($"The expression used in the ORDER BY clause is wrong.");
```

Comment:
Let's move the exception throwing to a dedicated method like `ThrowInvalidQuery(string message)`